### PR TITLE
Conditionally enable bitshift on ConstantinopleForkBlock

### PR DESCRIPTION
**Pull request by tcsiwula**
Conditionally enable bitshift on ConstantinopleForkBlock

### CODE REVIEW

File: libethcore/EVMSchedule.h [hunk 1]
```diff
@@ -33,6 +33,7 @@ struct EVMSchedule
 	bool haveDelegateCall = true;
 	bool eip150Mode = false;
 	bool eip158Mode = false;
+	bool eip145Mode = false;
```
Comment (gumb0) [inline libethcore/EVMSchedule.h]: We should better give it more descriptive name like `haveBitwiseShifting`

**Comment by gumb0** [comment]
Hey Tim, one more thing that needs to be done here is actually using the flag that you've added.

So the goal is to get rid of compile-time checks `#if EIP_145` in https://github.com/ethereum/cpp-ethereum/pull/4054 and insted check this flag in every shift instruction 
```
			if (!m_schedule->haveBitwiseShifting)
				throwBadInstruction();
```

File: libevm/VM.cpp [hunk 2]
```diff
@@ -653,6 +660,10 @@ void VM::interpretCases()
 
 		CASE(SAR)
 		{
+			// Pre-constantinople
+			if (!m_schedule->haveBitwiseShifting)
+				throwBadInstruction();
+
 			ON_OP();
 			updateIOGas();
```
Comment (axic) [inline libevm/VM.cpp]: There is a missed else-endif block below.

Comment (tcsiwula) [reply libevm/VM.cpp]: Thanks Alex :)

**Comment by axic** [comment]
Can you please squash some of these commits? :)

**Comment by tcsiwula** [comment]
Hoping this works ^_^

**Comment by axic** [comment]
Thanks!

**Pull request merged**